src/http_rest_api_handler.cpp

```diff
@@ -424,6 +383,56 @@ static Status validateContentFieldsEmptiness(KFSTensorInputProto* input) {
     return StatusCode::OK;
 }
 
+static bool isInputEmpty(::KFSRequest::InferInputTensor* input) {
+    if (input->datatype() == "FP32")
+        return input->contents().fp32_contents_size() == 0;
+    if (input->datatype() == "INT64")
+        return input->contents().int64_contents_size() == 0;
+    if (input->datatype() == "INT32")
+        return input->contents().int_contents_size() == 0;
+    if (input->datatype() == "INT16")
+        return input->contents().int_contents_size() == 0;
+    if (input->datatype() == "INT8")
+        return input->contents().int_contents_size() == 0;
+    if (input->datatype() == "UINT64")
+        return input->contents().uint64_contents_size() == 0;
+    if (input->datatype() == "UINT32")
+        return input->contents().uint_contents_size() == 0;
+    if (input->datatype() == "UINT16")
+        return input->contents().uint_contents_size() == 0;
+    if (input->datatype() == "UINT8")
+        return input->contents().uint_contents_size() == 0;
+    if (input->datatype() == "FP64")
+        return input->contents().fp64_contents_size() == 0;
+    if (input->datatype() == "BYTES")
+        return input->contents().bytes_contents_size() == 0;
+    return true;
+}
+
+static Status handleBinaryInput(const int& binary_input_size, size_t& binary_input_offset, const size_t& binary_inputs_size, const char* binary_inputs, ::KFSRequest::InferInputTensor* input) {
+    if (binary_input_offset + binary_input_size > binary_inputs_size) {
+        SPDLOG_DEBUG("Binary inputs size exceeds provided buffer size {}", binary_inputs_size);
+        return StatusCode::REST_BINARY_BUFFER_EXCEEDED;
+    }
+    auto status = parseBinaryInput(input, binary_input_size, binary_inputs + binary_input_offset);
+    if (!status.ok()) {
+        SPDLOG_DEBUG("Parsing binary inputs failed");
+        return status;
+    }
+    binary_input_offset += binary_input_size;
+    return StatusCode::OK;
+}
+
+static Status calculateBinaryDataSize(::KFSRequest::InferInputTensor* input, size_t& binary_data_size) {
+    auto element_size = KFSDataTypeSize(input->datatype());
+    size_t elements_number = 1;
+    for (auto dim : input->shape()) {
```

Comment:
You can use stl to calculate elements number in one line: https://github.com/openvinotoolkit/model_server/blob/main/src/customnodesession.cpp#L197